api/src/main/java/org/openmrs/module/bedmanagement/dao/BedManagementDao.java

```diff
@@ -50,4 +52,61 @@
     Bed getLatestBedByVisit(String visitUuid);
 
     List<BedTag> getAllBedTags();
+
+    /**
+     * Get all admission locations
+     * @param locations {@link List<Location>}
+     * @return {@link List<AdmissionLocation>}
+     */
+    List<AdmissionLocation> getAdmissionLocations(List<Location> locations);
+
+    /**
+     * Get admission location by location
+     *
+     * @param location {@link Location}
+     * @return {@link AdmissionLocation} return null if not exist
+     */
+    AdmissionLocation getAdmissionLocationByLocation(Location location);
```

Comment:
@sanishmaharjan is this the same as `getLayoutForWard(Location)`?

Reply:
Can I ask you again to refactor this as `getAdmissionLocationForLocation` and update the Javadoc as:
```
/**
  * Get the admission location for a given location.
  * Locations marked with the appropriate tag are <i>admission</i> locations. This method
  * returns the admission location, if applicable, when given a candidate location.
  *
  * @param location {@link Location}
  * @return The {@link AdmissionLocation} or null if the location is not an admisson location.
  */
```

Reply:
No, these are different in implementations. <b>getLayoutForWard(Location)</b> function only return bedLayouts property of child admission location But <b>getAdmissionLocationByLocation(Location location)</b> return admission location with all its properties with bedLayouts of its and its child admission locations.

Reply:
I did a `grep` on your branch and `getLayoutForWard` isn't used anywhere. Is there any reason to keep this?

Also the co-existence of both methods is confusing from an API standpoint. Because the possibility would exist to obtain two different states of the _same_ admission location. One with and one without nested bed layout information. Can you let me know your thoughts here?

Reply:
Nested bed layout for supporting multiple layer admission location hierarchies of admission location.

I have not remove existing function code in this pull request so getLayoutForWard exist in my branch that not used anywhere. For now, I have removed getLayoutForWard function.